src/solvers/dg_curved/containers_2d.jl

```diff
@@ -0,0 +1,91 @@
+function init_elements!(elements, mesh::CurvedMesh{2}, basis::LobattoLegendreBasis{T, NNODES}) where {T, NNODES}
+  @unpack faces = mesh
+  @unpack node_coordinates, left_neighbors, metric_terms, inverse_jacobian = elements
+
+  linear_indices = LinearIndices(size(mesh))
+
+  # Calculate inverse Jacobian and node coordinates
+  for cell_x in 1:size(mesh, 1), cell_y in 1:size(mesh, 2)
+    element = linear_indices[cell_x, cell_y]
+
+    calc_node_coordinates!(node_coordinates, element, cell_x, cell_y, mesh, basis)
+
+    calc_metric_terms!(metric_terms, element, node_coordinates, basis)
+
+    calc_inverse_jacobian!(inverse_jacobian, element, metric_terms)
+  end
+
+  initialize_neighbor_connectivity!(left_neighbors, mesh, linear_indices)
+
+  return nothing
+end
+
+
+function calc_node_coordinates!(node_coordinates, element,
+                                cell_x, cell_y,
+                                mesh::CurvedMesh{2},
+                                basis::LobattoLegendreBasis{T, NNODES}) where {T, NNODES}
+  @unpack nodes = basis
+
+  # Get cell length in reference mesh
+  dx = 2 / size(mesh, 1)
+  dy = 2 / size(mesh, 2)
+
+  # Calculate node coordinates of reference mesh
+  cell_x_offset = -1 + (cell_x-1) * dx + dx/2
+  cell_y_offset = -1 + (cell_y-1) * dy + dy/2
+
+  for j in 1:NNODES, i in 1:NNODES
+    # node_coordinates are the mapped reference node_coordinates
+    node_coordinates[:, i, j, element] .= transfinite_mapping(cell_x_offset + dx/2 * nodes[i],
+                                                              cell_y_offset + dy/2 * nodes[j], mesh)
+  end
+end
```

Comment:
```suggestion
function calc_node_coordinates!(node_coordinates, element,
                                cell_x, cell_y,
                                mesh::CurvedMesh{2},
                                basis::LobattoLegendreBasis)
  @unpack nodes = basis

  # Get cell length in reference mesh
  dx = 2 / size(mesh, 1)
  dy = 2 / size(mesh, 2)

  # Calculate node coordinates of reference mesh
  cell_x_offset = -1 + (cell_x-1) * dx + dx/2
  cell_y_offset = -1 + (cell_y-1) * dy + dy/2

  for j in eachindex(nodes), i in eachindex(nodes)
    # node_coordinates are the mapped reference node_coordinates
    node_coordinates[:, i, j, element] .= transfinite_mapping(cell_x_offset + dx/2 * nodes[i],
                                                              cell_y_offset + dy/2 * nodes[j], mesh)
  end
end
```
Alternatively, we could also define `eachnode(basis::LobattoLegendreBasis)` appropriately. Maybe that would be even better